streams/src/test/java/org/apache/kafka/streams/integration/SuppressionIntegrationTest.java

```diff
@@ -0,0 +1,319 @@
+        }
+    }
+
+    private long scaledTime(final long unscaledTime) {
+        return SCALE_FACTOR * unscaledTime;
+    }
+
+    private void produceSynchronously(final String topic, final List<KVT<String, String>> toProduce) {
+        final Properties producerConfig = mkProperties(mkMap(
+            mkEntry(ProducerConfig.CLIENT_ID_CONFIG, "anything"),
+            mkEntry(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, STRING_SERIALIZER.getClass().getName()),
+            mkEntry(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, STRING_SERIALIZER.getClass().getName()),
+            mkEntry(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers())
+        ));
+        try (final Producer<String, String> producer = new KafkaProducer<>(producerConfig)) {
+            // TODO: test EOS
+            //noinspection ConstantConditions
+            if (false) {
+                producer.initTransactions();
+                producer.beginTransaction();
+            }
+            final LinkedList<Future<RecordMetadata>> futures = new LinkedList<>();
+            for (final KVT<String, String> record : toProduce) {
+                final Future<RecordMetadata> f = producer.send(
+                    new ProducerRecord<>(topic, null, record.timestamp, record.key, record.value, null)
+                );
+                futures.add(f);
+            }
+            for (final Future<RecordMetadata> future : futures) {
+                try {
+                    future.get();
+                } catch (final InterruptedException | ExecutionException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+            // TODO: test EOS
+            //noinspection ConstantConditions
+            if (false) {
+                producer.commitTransaction();
+            }
+            producer.flush();
+        }
+    }
+
+    private void verifyOutput(final String topic, final List<KVT<String, Long>> expected) {
+        final List<ConsumerRecord<String, Long>> results = drainConsumerRecords(topic, STRING_DESERIALIZER, LONG_DESERIALIZER, expected.size());
+        if (results.size() != expected.size()) {
+            throw new AssertionError(printRecords(results) + " != " + expected);
+        }
+        final Iterator<KVT<String, Long>> expectedIterator = expected.iterator();
+        for (final ConsumerRecord<String, Long> result : results) {
+            final KVT<String, Long> expected1 = expectedIterator.next();
+            try {
+                compareKeyValueTimestamp(result, expected1.key, expected1.value, expected1.timestamp);
+            } catch (final AssertionError e) {
+                throw new AssertionError(printRecords(results) + " != " + expected, e);
+            }
+        }
+    }
+
+    private <K, V> List<ConsumerRecord<K, V>> drainConsumerRecords(final String topic, final Deserializer<K> keyDeserializer, final Deserializer<V> valueDeserializer, final int numRecords) {
```

Comment:
Same here, why not use one of the `IntegrationTestUtils.waitUtill...` methods for consuming records?

Reply:
Basically, the same explanation as above.

Reply:
Not sure if I understand. If it's a the same argument, could you not use `IntegrationTestUtils.waitUtill..` twice instead of duplicating the code?

Reply:
Ah, my apologies to @bbejeck . You are right. This is not the same argument as above, and the `waitUntilMinRecordsReceived` is fine here.